ledger/eval.go

```diff
@@ -1281,10 +1281,11 @@ transactionGroupLoop:
 				return ledgercore.StateDelta{}, err
 			}
 		}
-		err = eval.finalValidation()
-		if err != nil {
-			return ledgercore.StateDelta{}, err
-		}
+	}
+
+	err = eval.finalValidation()
```

Comment:
The move of `eval.finalValidation` outside the `validate` block ensures that we perform the Totals calculation in all cases. The `finalValidation` already has a test for the `eval. validate` inside to exclude the validation if that is not needed.